lib/cloudant.js

```diff
@@ -862,19 +884,25 @@ Cloudant.prototype.replaceById = function replaceById(
   var self = this;
   var mo = self.selectModel(model);
   var idName = self.idName(model);
-  data[idName] = id;
+  data[idName] = id.toString();
 
-  self.getCurrentRevision(model, id, function(err, rev) {
+  var replaceHandler = function(err, id) {
     if (err) return cb(err);
-    if (rev) data._rev = rev;
-    self._insert(model, data, function(err, id) {
+    mo.db.get(id, function(err, doc) {
       if (err) return cb(err);
-      mo.db.get(id, function(err, doc) {
-        if (err) return cb(err);
-        cb(null, self.fromDB(model, mo, doc));
-      });
+      cb(null, self.fromDB(model, mo, doc));
     });
-  });
+  };
+
+  if (!data._rev) {
+    self.getCurrentRevision(model, id, function(err, rev) {
```

Comment:
Then fix the DAO we must.
On the face of it, `replaceById` is one of those functions where the user might say "Ehh, I don't care about what's there. If it's there, get rid of it".
The trouble is that if someone were to update that document before the user does, how do we, the framework devs of LoopBack, know whether or not a user doesn't want to halt at that point and rethink their choice?

For the same reason as `updateOrCreate` and `replaceOrCreate`, I don't think that we should have `replaceById` altering data with mismatched `_rev` values.